lib/string-checker.js

```diff
@@ -130,7 +130,12 @@ StringChecker.prototype = {
 
         this._configuredRules.forEach(function(rule) {
             errors.setCurrentRule(rule.getOptionName());
-            rule.check(file, errors);
+
+            try {
```

Comment:
I was pushing for the opmization because i was under impression we would take hit. If there is nothing hurting the performance then its all good by me.

#1043 is not about perf improvements by about perf monitoring, we would need address every bottleneck on case-by-case basis